pkg/logger/logger.go

```diff
@@ -115,30 +120,51 @@ func New(flag int) *Logger {
 
 var std = New(LstdFlags)
 
-// Output a standard log template with a simple message.
-// Write a final newline at the end of every message.
-func (l *Logger) Output(calldepth int, message string) {
-	if !l.stdEnabled {
-		return
-	}
-
+func (l *Logger) formatLogMessage(calldepth int, message string) []byte {
 	now := time.Now()
 	file := "???:0"
 
 	if l.flag&(Lshortfile|Llongfile) != 0 {
 		file = l.GetFileLineString(calldepth + 1)
 	}
 
-	l.mu.Lock()
-	defer l.mu.Unlock()
-
-	l.stdLogTemplate.Execute(l.writer, stdLogMessageData{
+	var logBuff = new(bytes.Buffer)
+	l.stdLogTemplate.Execute(logBuff, stdLogMessageData{
 		Timestamp: FormatTimestamp(now),
 		File:      file,
 		Message:   message,
 	})
+	logBuff.Write([]byte("\n"))
 
-	l.writer.Write([]byte("\n"))
+	return logBuff.Bytes()
+}
+
+// Output a standard log template with a simple message to default output channel.
+// Write a final newline at the end of every message.
+func (l *Logger) Output(calldepth int, message string) {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+	if !l.stdEnabled {
+		return
+	}
+	msg := l.formatLogMessage(calldepth, message)
+	l.writer.Write(msg)
+}
+
+// OutputErr a standard log template with a simple message to the error output channel.
+// Write a final newline at the end of every message.
+func (l *Logger) OutputErr(calldepth int, message string) {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+	if !l.stdEnabled {
+		return
+	}
+	msg := l.formatLogMessage(calldepth, message)
+	if l.errToStdout {
+		l.writer.Write(msg)
+	} else {
+		l.errWriter.Write(msg)
+	}
```

Comment:
I wonder if instead of doing this, `SetErrToStdout` just does `l.errWriter = l.writer`, WDYT? We wouldn't actually need `errToStdout` as a member of the logger struct at all